File: chapter11/listing_11_22-todo-scala-joint-compilation/repository/src/main/scala/com/manning/gia/todo/repository/ToDoRepository.scala
package com.manning.gia.todo.repository

import com.manning.gia.todo.model.ToDoItem

trait ToDoRepository {
    def findAll(): java.util.List[ToDoItem]
    def findAllActive(): java.util.List[ToDoItem]
    def findAllCompleted(): java.util.List[ToDoItem]
    def findById(id: java.lang.Long): ToDoItem
    def insert(toDoItem: ToDoItem): java.lang.Long
    def update(toDoItem: ToDoItem): Unit
    def delete(toDoItem: ToDoItem): Unit
}

File: chapter11/listing_11_10-12-todo-grunt-integration/repository/src/main/java/com/manning/gia/todo/repository/ToDoRepository.java
package com.manning.gia.todo.repository;

import com.manning.gia.todo.model.ToDoItem;

import java.util.List;

public interface ToDoRepository {
    List<ToDoItem> findAll();

    List<ToDoItem> findAllActive();

    List<ToDoItem> findAllCompleted();

    ToDoItem findById(Long id);

    Long insert(ToDoItem toDoItem);

    void update(ToDoItem toDoItem);

    void delete(ToDoItem toDoItem);
}